static/js/filter.js

```diff
@@ -497,6 +497,17 @@ Filter.prototype = {
         return new Filter(terms);
     },
 
+    filter_with_new_params: function (params) {
+        const terms = this._operators.map(term => {
+            const new_term = { ...term };
+            if (new_term.operator === params.operator && !new_term.negated) {
+                new_term.operand = params.operand;
+            }
+            return new_term;
+        });
+        return new Filter(terms);
+    },
```

Comment:
This can just replace `filter_with_new_topic`.  Can you extract this change as a prep commit that renames this and expands its functionality?  Then we can merge that ahead of the rest.

Reply:
Sure. :-)

Reply:
@timabbott how do I call this new PR??

Reply:
I created this PR: https://github.com/zulip/zulip/pull/14109

Reply:
With `git rebase -i`, you can add a commit at the start of your PR for this sort of thing, but I suppose a new PR works too.